### DISCUSSION TITLE
Hello, how can I achieve fast simulation in the OperationDesk by setting the time?

### DISCUSSION THREAD

**Discussion by SHMay100G** [openTCS/opentcs#108] [General]
Hello, how can I achieve fast simulation in the OperationDesk by setting the time?

**Comment by swltr** [comment]
You can set `virtualvehicle.simulationTimeFactor` in the kernel configuration (see [the User's Guide](https://opentcs.org/docs/5.10/user/opentcs-users-guide.html#_virtual_vehicle_configuration_entries)).

**Reply by SHMay100G** [reply]
Thank you very much! I am now able to conduct fast simulations. Additionally, I have another question: How can I dynamically change virtualvehicle.simulationTimeFactor in real-time during the fast simulation process?

**Reply by swltr** [reply]
The documentation at the URL linked to before says "Trigger for changes to be applied: instantly", so when you change the entry in the file, it should be picked up automatically, without restarting the software.

For changing the setting programmatically during runtime, there is no API.

Also please keep in mind that openTCS is not a simulation tool. The virtual vehicle is merely meant to test-drive your plant model, not more.
